tests/ops/functions/conftest.py

```diff
@@ -14,74 +14,125 @@
-    qml.ops.Sum,
-    qml.BasisState,
-    qml.ControlledQubitUnitary,
-    qml.QubitStateVector,
-    qml.QubitChannel,
-    qml.MultiControlledX,
-}
+_INSTANCES_TO_TEST = [
+    qml.sum(qml.PauliX(0), qml.PauliZ(0)),
+    qml.BasisState([1], wires=[0]),
+    qml.ControlledQubitUnitary(np.eye(2), control_wires=1, wires=0),
+    qml.QubitStateVector([0, 1], wires=0),
+    qml.QubitChannel([np.array([[1, 0], [0, 0.8]]), np.array([[0, 0.6], [0, 0]])], wires=0),
+    qml.MultiControlledX(wires=[0, 1]),
+    qml.Projector([1], 0),  # the state-vector version is already tested
+    qml.SpecialUnitary([1, 1, 1], 0),
+    qml.IntegerComparator(1, wires=[0, 1]),
+    qml.PauliRot(1.1, "X", wires=[0]),
+    qml.StatePrep([0, 1], 0),
+    qml.PCPhase(0.27, dim=2, wires=[0, 1]),
+    qml.BlockEncode([[0.1, 0.2], [0.3, 0.4]], wires=[0, 1]),
+    qml.adjoint(qml.PauliX(0)),
+    qml.adjoint(qml.RX(1.1, 0)),
+    Tensor(qml.PauliX(0), qml.PauliX(1)),
+    qml.Hamiltonian([1.1, 2.2], [qml.PauliX(0), qml.PauliZ(0)]),
+    qml.s_prod(1.1, qml.RX(1.1, 0)),
+    qml.prod(qml.PauliX(0), qml.PauliY(1), qml.PauliZ(0)),
+    qml.ctrl(qml.RX(1.1, 0), 1),
+    qml.exp(qml.PauliX(0), 1.1),
+    qml.ops.Evolution(qml.PauliX(0), 5.2),
+    qml.QutritBasisState([1, 2, 0], wires=[0, 1, 2]),
+]
+"""Valid operator instances that could not be auto-generated."""
+
+
+_INSTANCES_TO_FAIL = [
+    (
+        qml.SparseHamiltonian(qml.Hamiltonian([1.1], [qml.PauliX(0)]).sparse_matrix(), [0]),
+        AssertionError,  # each data element must be tensorlike
+    ),
+    (
+        qml.PauliError("X", 0.5, wires=0),
+        AssertionError,  # each data element must be tensorlike
+    ),
+    (
+        qml.THermitian(np.eye(3), wires=0),
+        AssertionError,  # qutrit ops fail validation
+    ),
+    (
+        qml.ops.qubit.special_unitary.TmpPauliRot(1.1, "X", [0]),
+        AssertionError,  # private type with has_matrix=False despite having one
+    ),
+    (
+        qml.ops.Conditional(qml.measure(1), qml.S(0)),
+        AssertionError,  # needs flattening helpers to be updated, also cannot be pickled
+    ),
+    (
+        qml.prod(qml.RX(1.1, 0), qml.RY(2.2, 0), qml.RZ(3.3, 1)),
+        DiagGatesUndefinedError,  # has_diagonalizing_gates should be False
+    ),
+    (
+        qml.Identity(0),
+        MatrixUndefinedError,  # empty decomposition, matrix differs from decomp's matrix
+    ),
+    (
+        qml.GlobalPhase(1.1),
+        MatrixUndefinedError,  # empty decomposition, matrix differs from decomp's matrix
+    ),
+    (
+        qml.pulse.ParametrizedEvolution(qml.PauliX(0) + (lambda p, t: p * t) * qml.PauliZ(0)),
+        PickleError,  # Can't pickle, binding parameters fails, and more
+    ),
+    (
+        qml.pow(qml.IsingXX(1.1, [0, 1]), 2.5),
+        PickleError,  # Can't pickle, binding parameters fails, and more
+    ),
+    (
+        qml.resource.FirstQuantization(1, 2, 1),
+        IndexError,  # too many qubits for validation function
```

Comment:
We could skip the wire mapping test if the operator has more than 26 wires?

Reply:
done